src/Services/Air/Air.js

```diff
@@ -134,19 +135,28 @@ module.exports = (settings) => {
     },
 
     ticket(options) {
-      return this.getPNR(options).then((booking) => {
-        const ticketParams = Object.assign({}, options, {
-          ReservationLocator: booking.uapi_reservation_locator,
-        });
-        return service.ticket(ticketParams)
-          .then(result => result, (err) => {
-            if (err instanceof AirRuntimeError.TicketingFoidRequired) {
-              return this.getPNR(options)
-                .then(updatedBooking => service.foid(updatedBooking))
-                .then(() => service.ticket(ticketParams));
-            }
-            return Promise.reject(err);
-          });
+      return retry({ retries: 3 }, (again, number) => {
+        if (settings.debug && number > 1) console.log(`ticket ${options.pnr} retry number ${number}`);
+        return (options.ReservationLocator ? Promise.resolve(options) :
+          this.getPNR(options).then((booking) => {
+            const ticketParams = Object.assign({}, options, {
+              ReservationLocator: booking.uapi_reservation_locator,
+            });
+            return ticketParams;
+          })
+        ).then(
+          ticketParams => service.ticket(ticketParams)
+            .then(result => result, (err) => {
```

Comment:
Please, use `catch`, instead of `then` with two params.
If `catch` will have nothing to catch, then result will be returned on its own.

Reply:
Was like it in the original, will rewrite.

Reply:
Thanks!